src/utils/cache.py

```diff
@@ -9,6 +13,15 @@ def caching_decorator(func):
         cached_func = functools.lru_cache(*args, **kwargs)(func)
 
         def wrapper(*args, **kwargs):
+
+            # If lru_cache is on contract
+            # Do not cache any requests with relative blocks
+            # Like 'latest', 'earliest', 'pending', 'safe', 'finalized' or if default block provided
```

Comment:
But if we pass the value "latest" as an argument it will be still cached, won't it?